modules/kotest/src/main/kotlin/org/testcontainers/containers/StartablePerSpecListener.kt

```diff
@@ -0,0 +1,34 @@
+package org.testcontainers.containers
+
+import io.kotest.core.listeners.TestListener
+import io.kotest.core.spec.Spec
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.testcontainers.lifecycle.Startable
+
+
+/**
+ * [StartablePerSpecListener] starts the given [startable] before execution of any test in the spec
+ * and stops after execution of all tests.
+ *
+ * [startable] can any of [GenericContainer] [DockerComposeContainer] [LocalStackContainer] etc.
+ *
+ * This should be use when you want to a single container for all test in a single test class.
+ *
+ * @see
+ * [StartablePerTestListener]
+ * */
+
+internal class StartablePerSpecListener(private val startable: Startable) : TestListener {
+    override suspend fun beforeSpec(spec: Spec) {
+        withContext(Dispatchers.IO) {
+            startable.start()
```

Comment:
`TestLifecycleAware` support is still missing...

Reply:
TestLifecycleAware has two methods beforeTest and afterTest, as whereas SpecListener are for runnig before and after the entire spec(Test class). Thats the reason i have not added it there.

Reply:
please see other integrations

Reply:
@ashishkujoy As I understand it, if the startable is a TestLifecycleAware, then we should call beforeTest and afterTest on it, as the container may need to do "other things" between tests.

Reply:
The documentation needs to be corrected:
_This should be use when you want to a single container for all test in a single test class._
to
_This listener should be used when you want to use a single container for all tests in a single spec class._

Reply:
You could store them in a variable in the beforeSpec callback.

Reply:
support added

Reply:
@ashishkujoy please *do not* resolve conversations started by others. e.g. this one remains unresolved, in fact.

Reply:
> That would be difficult because the before and after test functions in TestLifecylceAware takes TestDescription as parameter which in itself needs testId and testName and these informations are not present in Spec class which we receive in beforeSpec and afterSpec callback.

something to improve in kotest? Other test frameworks don't have this problem.